.github/workflows/Github_pytest.yml

```diff
@@ -56,22 +58,43 @@ jobs:
         uses: ./.github/actions/windows_install
       - name: Install python dependencies
         uses: ./.github/actions/pip_installation
-      - name: Serial C tests
+
+      # This is split into many tests as the test raising an error must be the last line in a multi-line powershell command
+      - name: Serial C tests run in parallel
+        run: |
+          python -m pytest -n auto -rx -m "not (parallel or xdist_incompatible) and c" --ignore=symbolic --ignore=ndarrays -x
+        working-directory: ./tests
+      - name: Serial C tests run in serial
+        run: |
+          pyccel-clean
+          python -m pytest -rx -m "xdist_incompatible and not parallel and c" --ignore=symbolic --ignore=ndarrays
+        working-directory: ./tests
+      - name: Serial fortran tests and non-tagged tests run in parallel
+        run: |
+          python -m pytest -n auto -rx -m "not (parallel or xdist_incompatible) and not (c or python)" --ignore=symbolic --ignore=ndarrays
+        working-directory: ./tests
+      - name: Serial fortran tests and non-tagged tests run in serial
         run: |
-          python -m pytest -rx -m "not parallel and c" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          pyccel-clean
+          python -m pytest -rx -m "xdist_incompatible and not parallel and not (c or python)" --ignore=symbolic --ignore=ndarrays
         working-directory: ./tests
-      - name: Serial fortran tests and non-tagged tests
+      - name: Serial Python tests run in parallel
         run: |
-          python -m pytest -rx -m "not parallel and not (c or python)" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          python -m pytest -n auto -rx -m "not (parallel or xdist_incompatible) and python" --ignore=symbolic --ignore=ndarrays
         working-directory: ./tests
-      - name: Serial Python tests
+      - name: Serial Python tests cleanup
         run: |
-          python -m pytest -rx -m "not parallel and python" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          pyccel-clean
+          #python -m pytest -rx -m "xdist_incompatible and not parallel and python" --ignore=symbolic --ignore=ndarrays
```

Comment:
Should we remove this commented line?

Reply:
See above

Reply:
I see, but surely this line does not belong to step "Serial Python tests cleanup"?

Reply:
No, but I'm not sure that I can create an action which does nothing. I.e.
```
- name: Serial Python tests run in parallel
   run : |
       #python -m pytest -rx -m "xdist_incompatible and not parallel and python" --ignore=symbolic --ignore=ndarrays
   working-director : tests
```

Reply:
I think it's OK (and clearer) if you comment out multiple lines.

Reply:
Fixed in 54cb3c8